### PR TITLE
zkEVM docs review 

### PR DESCRIPTION
**Pull request by EmpieichO**
zkEVM docs review - as per ticket https://polygon.atlassian.net/browse/DOCS-9 

Since this is a bit chunky, I will continue with further review of the zkEVM docs after these commits have been merged.

### CODE REVIEW

File: docs/zkEVM/architecture/high-level/overview.md [hunk 1]
```diff
@@ -1,7 +1,5 @@
 <!-- https://excalidraw.com/#json=JKZp9QEihifF_B7Z41Dfv,FVNhqQKi9PA1jM0kzUoCsQ" -->
 
-## Intro
-
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/high-level/overview.md:4]: don't delete this, it's a useful right-hand-side toc element, rename it if you have to

File: docs/zkEVM/architecture/protocol/incentive-mechanism.md [hunk 2]
```diff
@@ -71,8 +71,8 @@ function _updateBatchFee(uint64 newLastVerifiedBatch) internal
 
 The admin defines two storage variables that are used to **tune the fee adjustment function**:
 
-- `veryBatchTimeTarget`: it is **the targeted time of the verification of a batch**, so the `batchFee` variable will be updated to achieve this target, and
-- `multiplierBatchFee`: it is the batch fee multiplier, with 3 decimals that **ranges from 1000 to 1024**.
+- `veryBatchTimeTarget`, which is **the targeted time of the verification of a batch**, so the `batchFee` variable is updated to achieve this target, and
+- `multiplierBatchFee`, which is the batch fee multiplier, with 3 decimals **ranging from 1000 to 1024**.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/incentive-mechanism.md]:
```suggestion
- `multiplierBatchFee`, which is the batch fee multiplier, with 3 decimals ranging from 1000 to 1024.
```

File: docs/zkEVM/architecture/high-level/overview.md [hunk 3]
```diff
@@ -1,7 +1,5 @@
 <!-- https://excalidraw.com/#json=JKZp9QEihifF_B7Z41Dfv,FVNhqQKi9PA1jM0kzUoCsQ" -->
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/high-level/overview.md:2]:
```suggestion

## Intro
```

File: docs/zkEVM/architecture/protocol/index.md [hunk 4]
```diff
@@ -1,49 +1,47 @@
...
+It takes three main components of the Polygon zkEVM protocol to enable transaction finality while ensuring the correctness of state transitions:
 
-The three main components of zkEVM protocol are:
-
-- Trusted sequencer.
-- Trusted aggregator.
-- Consensus contract (PolygonZkEVM.sol, deployed on L1).
+- The trusted sequencer.
+- The trusted aggregator.
+- The consensus contract (`PolygonZkEVM.sol`, deployed on L1).
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/index.md:13]: This is no longer the consensus contract.

File: docs/zkEVM/architecture/protocol/malfunction-resistance/aggregator-resistance.md [hunk 5]
```diff
@@ -15,14 +15,14 @@ function verifyBatches(
 ) public ifNotEmergencyState
 ```
 
-As previously stated, the `verifyBatches` function accepts the same arguments as the `trustedVerifyBatches` function. However, the `verifyBatches` function adds two additional constraints for a sequence to be aggregated, as well as a new L2 State stage called Pending State.
+As previously stated, the `verifyBatches` function accepts the same arguments as the `trustedVerifyBatches` function. However, the `verifyBatches` function adds two additional constraints for a sequence to be aggregated, as well as a new L2 state stage called Pending state.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/malfunction-resistance/aggregator-resistance.md]:
```suggestion
As previously stated, the `verifyBatches` function accepts the same arguments as the `trustedVerifyBatches` function. However, the `verifyBatches` function adds two additional constraints for a sequence to be aggregated, as well as a new L2 pending state.
```

File: docs/zkEVM/architecture/protocol/transaction-life-cycle/batch-sequencing.md [hunk 6]
```diff
@@ -29,21 +29,21 @@ The contract's public constant, `MAX_TRANSACTIONS_BYTE_LENGTH`, determines the m
 
 Similarly, the number of batches in a sequence is limited by the contract's public constant `MAX_VERIFY_BATCHES` (1000). The batches array must contain at least one batch and no more than the value of the constant `MAX_VERIFY_BATCHES`.
 
-Only the trusted sequencer's Ethereum account can access the `sequencedBatches` mapping. It is also necessary that the contract not be in an emergency state.
+Only the trusted sequencer's Ethereum account can access the `sequencedBatches` mapping. The contract must not be in an emergency state.
 
-The function call will be reverted if the above conditions are not met.
+The function call is reverted if the above conditions are not met.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/transaction-life-cycle/batch-sequencing.md]:
```suggestion
The function call reverts if the above conditions are not met.
```

File: docs/zkEVM/architecture/protocol/transaction-life-cycle/batch-sequencing.md [hunk 7]
```diff
@@ -29,21 +29,21 @@ The contract's public constant, `MAX_TRANSACTIONS_BYTE_LENGTH`, determines the m
...
+- The timestamp of the block must be greater or equal to that of the last block (of a sequenced batch), but less than or equal to the timestamp of the block where the sequencing L1 transaction is executed. All blocks must be ordered by time.
 
-If one batch is not valid, the transaction reverts, discarding the entire sequence. Otherwise, if all batches to be sequenced are valid, the sequencing process will continue.
+If one block is not valid, the transaction reverts, discarding the entire sequence. Otherwise, if all blocks in the batches that are to be sequenced are valid, the sequencing process continues as normal.
 
-A storage variable called `lastBatchSequenced` is used as a batch counter, and it is thus incremented each time a batch is sequenced. It gives a specific index number to each batch that will be used as a position value in the batch chain.
+A storage variable called `lastBatchSequenced` is used as a batch counter, and it is thus incremented each time a batch is sequenced. It gives a specific index number to each batch that is to be used as a position value in the batch chain.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/transaction-life-cycle/batch-sequencing.md]:
```suggestion
A storage variable called `lastBatchSequenced` is used as a batch counter, and it is thus incremented each time a batch is sequenced. It gives a specific index number to each batch that is used as a position value in the batch chain.
```

File: docs/zkEVM/architecture/protocol/transaction-life-cycle/batch-sequencing.md [hunk 8]
```diff
@@ -93,29 +93,29 @@ struct SequencedBatchData {
 - `sequencedTimestamp` is the timestamp of the block where the sequencing L1 transaction is executed.
 - `previousLastBatchSequenced` is the index of the last sequenced batch before the first batch of the current sequence (i.e., the last batch of the previous sequence).
 
-The index number of the last batch in the sequence is used as key and the `SequencedBatchData` struct is used as value when the sequence will be entered into `sequencedBatches` mapping.
+The index number of the last batch in the sequence is used as key and the `SequencedBatchData` struct is used as value when the sequence is entered into `sequencedBatches` mapping.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/transaction-life-cycle/batch-sequencing.md]:
```suggestion
The index number of the last batch in the sequence is the key, and the `SequencedBatchData` struct is the value, in the `sequencedBatches` mapping.
```

File: docs/zkEVM/architecture/protocol/transaction-life-cycle/batch-sequencing.md [hunk 9]
```diff
@@ -93,29 +93,29 @@ struct SequencedBatchData {
...
 
-As previously stated, the hash digest will be a commitment of the entire batch chain. Batch indices also commit useful information like the number of batches in the sequence and their position in the batch chain. The timestamp anchors the sequence to a specific point in time.
+As previously stated, the hash digest becomes a commitment of the entire batch chain. Batch indices also commit useful information like the number of batches in the sequence and their position in the batch chain. The timestamp anchors the sequence to a specific point in time.
 
-The data availability of the L2 transactions is guaranteed because the data of each batch can be recovered from the calldata of the sequencing transaction, which is not part of the contract storage but is part of the L1 State.
+The data availability of the L2 transactions is guaranteed because the data of each batch can be recovered from the calldata of the sequencing transaction, which is not part of the contract storage but is part of the L1 state.
 
-Finally a `SequenceBatches` event will be emitted.
+Finally a `SequenceBatches` event is emitted.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/transaction-life-cycle/batch-sequencing.md]:
```suggestion
Finally a `SequenceBatches` event emits.
```

File: docs/zkEVM/architecture/protocol/zkevm-bridge/exit-tree.md [hunk 10]
```diff
@@ -1,50 +1,50 @@
-The zkEVM's Bridge SC utilizes a special Merkle Tree called Exit Tree for each of the networks participating in the communication or asset-exchange.
+The zkEVM's bridge SC utilizes a special Merkle tree called Exit Tree for each of the networks participating in the communication or asset-exchange.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/zkevm-bridge/exit-tree.md]:
```suggestion
The zkEVM's bridge SC utilizes a special Merkle tree called an exit tree for each of the networks participating in the communication or asset-exchange.
```

File: docs/zkEVM/architecture/protocol/zkevm-bridge/exit-tree.md [hunk 11]
```diff
@@ -1,50 +1,50 @@
...
 
-The term Exit Tree refers to an append-only Sparse Merkle Tree (SMT) whose leaf nodes record information about assets being transferred out of the network. Exit Trees of depth 32 are used by Polygon zkEVM.
+The term `exit tree refers to an append-only Sparse Merkle tree (SMT) whose leaf nodes record information about assets being transferred out of the network. Exit trees of depth 32 are used by Polygon zkEVM.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/zkevm-bridge/exit-tree.md]:
```suggestion
The term exit tree refers to an append-only sparse Merkle tree (SMT) whose leaf nodes record information about assets being transferred out of the network. Exit trees of depth 32 are used by Polygon zkEVM.
```

File: docs/zkEVM/architecture/protocol/zkevm-bridge/exit-tree.md [hunk 12]
```diff
@@ -1,50 +1,50 @@
...
+- uint8 leafType: [0] asset, [1] message.
+- int32 originNetwork: Origin network ID, where the original asset belongs.
+- address originAddress: If `leafType = 0`, Origin network token address (`0x0000...0000`) is reserved for ether. If `leafType = 1`, `msg.sender` of the message.
+- uint32 destinationNetwork: Bridging destination network ID.
+- address destinationAddress: Address that receives the bridged asset in the destination network.
+- uint256 amount: Amount of tokens/ether to bridge.
+- bytes32 metadataHash: Hash of the metadata. This metadata contains information about asset transferred or the message payload.
 
-When a user commits to transferring assets from one network to another, the Bridge SC must add an exit leaf to that network's Exit Tree.
+When a user commits to transferring assets from one network to another, the bridge SC must add an exit leaf to that network's exit tree.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/zkevm-bridge/exit-tree.md]:
```suggestion
When a user commits to transferring assets from one network to another, the bridge contract adds an exit leaf to that network's exit tree.
```

File: docs/zkEVM/architecture/protocol/zkevm-bridge/exit-tree.md [hunk 13]
```diff
@@ -55,17 +55,17 @@ Amount: 145
...
+To claim the bridged assets on the destination L2 network, the global exit root is verified using a Merkle proof. This proof allows one to confirm whether an exit leaf, containing information about assets being bridged to L2, is represented in the global exit tree by the corresponding L1 exit tree root.
 
-![Updating L1 Exit Tree and the Global Exit Root](../../../../img/zkEVM/03pzb-exit-leaf-add-L1-L2.png)
+![Updating L1 exit tree and the global exit root](../../../../img/zkEVM/03pzb-exit-leaf-add-L1-L2.png)
 
 ### Transfer from rollup L2 to L1
 
-It is possible for the transfer to go from an L2 Rollup to the Mainnet L1. In this case the same procedure as in the above example is followed, except in the reverse direction.
+Transfers can also occur from an L2 Rollup to the Mainnet L1. In this scenario, the same procedure outlined in the previous example is followed, but in the reverse direction.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/zkevm-bridge/exit-tree.md]:
```suggestion
Transfers can also occur from an L2 Rollup to the L1 mainnet. In this scenario, the same procedure outlined in the previous example is followed, but in the reverse direction.
```

File: docs/zkEVM/architecture/protocol/zkevm-bridge/index.md [hunk 14]
```diff
@@ -1,45 +1,45 @@
-The ability of any blockchain to exchange data with other blockchains is critical, as it allows the blockchain to participate in an ecosystem of other blockchains; this is known as **interoperability**. Many interoperability-enabling solutions have been developed, each with their own set of advantages and disadvantages.
+The ability of a blockchain to exchange data with other blockchains is crucial for its participation in a broader ecosystem, a concept known as Interoperability. Various solutions have been developed to enable interoperability, each offering unique advantages and disadvantages.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/zkevm-bridge/index.md]:
```suggestion
The ability of a blockchain to exchange data with other blockchains is crucial for its participation in a broader ecosystem, a concept known as interoperability. Various solutions have been developed to enable interoperability, each offering unique advantages and disadvantages.
```

File: docs/zkEVM/architecture/protocol/zkevm-bridge/index.md [hunk 15]
```diff
@@ -1,45 +1,45 @@
...
-In L2 rollups such as the Polygon zkEVM, L1 smart contracts ensure proper management of L2 state transitions and data availability. As a result, with an appropriate L2 architecture design, both ends of the Bridge smart contract (SC) can be synchronized entirely through the smart contract's logic.
+In zk-rollups like the Polygon zkEVM, L1 smart contracts manage L2 state transitions and data availability. Consequently, with a well-designed L2 architecture, both ends of the bridge smart contract (SC) can be fully synchronized through the logic of the smart contract.
 
 ## zkEVM bridge schema
 
-Let's understand the zKEVM Bridge protocol design through an analogy. Make sure to check out the below illustration for easier understanding of the protocol.
+To better understand the zkEVM bridge protocol design, let's use an analogy. Refer to the illustration below for a clearer explanation of the protocol.
 
-![Polygon zkEVM Bridge Schema](../../../../img/zkEVM/01pzb-polygon-zkevm-schema.png)
+![Polygon zkEVM bridge Schema](../../../../img/zkEVM/01pzb-polygon-zkevm-schema.png)
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/zkevm-bridge/index.md]:
```suggestion
![Polygon zkEVM bridge schema](../../../../img/zkEVM/01pzb-polygon-zkevm-schema.png)
```

File: docs/zkEVM/architecture/protocol/zkevm-bridge/index.md [hunk 16]
```diff
@@ -1,45 +1,45 @@
...
 
-Once minting is complete, **the asset can be claimed by the user or recipient in the destination network (L2)**.
+Once minting is complete, the asset can be claimed by the user or recipient in the destination network (L2).
 
-It is also possible to perform the opposite operation. That is, after burning the Wrapped Token, the Bridge SC unlocks the original asset in the origin network.
+The operation can also be performed in reverse. After burning the Wrapped Token, the bridge smart contract unlocks the original asset on the origin network.
 
-As previously stated, there is the possibility of a third channel where the Bridge SC is used for cross-chain messaging. That is, sending data payloads from one network to another can be achieved using the Bridge SC's **Bridge** and **Claim** operations.
+As mentioned earlier, there is also a third option where the bridge smart contract facilitates cross-chain messaging. This allows data payloads to be sent from one network to another using the bridge smart contract's Bridge and Claim operations.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/zkevm-bridge/index.md]:
```suggestion
As mentioned earlier, there is also a third option where the bridge smart contract facilitates cross-chain messaging. This allows data payloads to be sent from one network to another using the bridge smart contract's bridge and claim operations.
```

File: docs/zkEVM/architecture/protocol/zkevm-bridge/index.md [hunk 17]
```diff
@@ -1,45 +1,45 @@
...
 
 As a result, as stated in this document, every transfer is protected by smart contracts.
 
 ## Features
 
-The Polygon zkEVM Bridge smart contract's main feature is the use of Exit Trees and the Global Exit Tree, with the Global Exit Tree Root serving as the primary source of state truth.
+The Polygon zkEVM bridge smart contract's main feature is the use of Exit Trees and the Global Exit Tree, with the Global Exit Tree Root serving as the primary source of state truth.
 
-The use of two distinct Global Exit Root managers for L1 and L2, as well as separate logic for the Bridge SC and each of these Global Exit Root managers, allows for extensive network interoperability.
+The use of two distinct Global Exit Root managers for L1 and L2, as well as separate logic for the bridge SC and each of these Global Exit Root managers, allows for extensive network interoperability.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/zkevm-bridge/index.md]:
```suggestion
The use of two distinct global exit root managers for L1 and L2, as well as separate logic for the bridge contract and each of these global exit root managers, allows for extensive network interoperability.
```

File: docs/zkEVM/architecture/protocol/zkevm-bridge/index.md [hunk 18]
```diff
@@ -1,45 +1,45 @@
...
 
-Second, the **Bridge SC generates wrapped tokens the first time a new token is added to the zkEVM network**. In addition, the **metadata of the ERC20 token, such as the name, decimal, or symbol, is added to the information contained in the leaf**.
+Second, the bridge SC generates wrapped tokens the first time a new token is added to the zkEVM network. In addition, the metadata of the ERC20 token, such as the name, decimal, or symbol, is added to the information contained in the leaf.
 
 As a result, as stated in this document, every transfer is protected by smart contracts.
 
 ## Features
 
-The Polygon zkEVM Bridge smart contract's main feature is the use of Exit Trees and the Global Exit Tree, with the Global Exit Tree Root serving as the primary source of state truth.
+The Polygon zkEVM bridge smart contract's main feature is the use of Exit Trees and the Global Exit Tree, with the Global Exit Tree Root serving as the primary source of state truth.
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/protocol/zkevm-bridge/index.md]:
```suggestion
The Polygon zkEVM bridge smart contract's main feature is the use of exit trees and the global exit tree, with the global exit tree root serving as the primary source of state truth.
```

File: docs/zkEVM/how-to/using-foundry.md [hunk 19]
```diff
@@ -183,6 +183,6 @@ The above command compiles and deploys the contract to the zkEVM Testnet. The ou
 
 - Copy the address of your newly deployed contract (i.e. the `Deployed to:` address as in the above example output).
 
-- Go to the [zkEVM Testnet Explorer](https://testnet-zkevm.polygonscan.com), and paste the address in the `Search by address` field.
+- Go to the [zkEVM Testnet Explorer](https://cardona-zkevm.polygonscan.com/), and paste the address in the `Search by address` field.
```
Comment (kmurphypolygon) [inline docs/zkEVM/how-to/using-foundry.md:186]:
```suggestion
- Go to the [zkEVM Testnet Explorer](https://cardona-zkevm.polygonscan.com/), and paste the address in the **Search by address** field.
```

Comment (kmurphypolygon) [reply docs/zkEVM/how-to/using-foundry.md:186]: UI elements are bold.

File: docs/zkEVM/how-to/using-hardhat.md [hunk 20]
```diff
@@ -34,19 +34,14 @@ This document is a guide on how to deploy a smart contract on the Polygon zkEVM
...
-- Select **Account details** and then click **Export private key**
-
-- Enter your Metamask password to reveal the private key
+    Select `Account details` and then click `Export private key`.
 
-- Copy the private key and paste it into the `.env.sample` file.
+    Enter your Metamask password to reveal the private key.
 
-</details>
+    Copy the private key and paste it into the `.env.sample` file.
```
Comment (kmurphypolygon) [inline docs/zkEVM/how-to/using-hardhat.md:44]: These should be numbered or bulleted, at the moment, they're kind of hanging in the air..

File: docs/zkEVM/how-to/verify-contract.md [hunk 21]
```diff
@@ -18,7 +18,7 @@ After successfully compiling a smart contract, follow the next steps to verify y
 
 6. Click the **Verify and Publish** button.
 
-7. There are 3 options to provide the Contract's code. We will be diving into the following two options:
+7. There are 3 options to provide the Contract's code. We dive into the `Flattened source code` and `Standard input JSON` options.
```
Comment (kmurphypolygon) [inline docs/zkEVM/how-to/verify-contract.md:21]:
```suggestion
7. There are 3 options to provide the contract's code. We look at the **Flattened source code** and **Standard input JSON** options.
```

File: docs/zkEVM/how-to/verify-contract.md [hunk 22]
```diff
@@ -66,13 +65,13 @@ After successfully compiling a smart contract, follow the next steps to verify y
...
 
-Once you paste the contents of the newly created `.sol` file to the _Enter the Solidity Contract_ field, the **Verify & Publish** button will be active.
+Once you paste the contents of the newly created `.sol` file to the _Enter the Solidity Contract_ field, the **Verify & Publish** button becomes active.
 
 Click on **Verify & Publish** to verify your deployed smart contract.
 
 ## Verify using Remix
 
-We will be using the ready-made `Storage.sol` contract in Remix. Compile the contract and follow the steps provided below.
+We use ready-made `Storage.sol` contract in Remix. Compile the contract and follow the steps provided below.
```
Comment (kmurphypolygon) [inline docs/zkEVM/how-to/verify-contract.md]:
```suggestion
We use the ready-made `Storage.sol` contract in Remix. Compile the contract and follow the steps provided below.
```

File: mkdocs.yml [hunk 23]
```diff
@@ -94,7 +94,7 @@ nav:
         - Deploy a contract with Hardhat: zkEVM/how-to/using-hardhat.md
         - Verify a contract: zkEVM/how-to/verify-contract.md
       - Architecture high-level:
-        - Overview: zkEVM/architecture/high-level/overview.md
+        - Architectural overview: zkEVM/architecture/high-level/overview.md
```
Comment (kmurphypolygon) [inline mkdocs.yml:97]: Why do you want to repeat the word from the menu header?

**Review by kmurphypolygon — APPROVED** [review]
Looks good, a few suggestions.

File: docs/zkEVM/architecture/high-level/overview.md [hunk 24]
```diff
@@ -1,7 +1,5 @@
...
 
-## Intro
```
Comment (kmurphypolygon) [inline docs/zkEVM/architecture/high-level/overview.md:3]: yeah, this should go :)

File: docs/zkEVM/architecture/protocol/index.md (hunk 4 above)
Comment (EmpieichO) [reply docs/zkEVM/architecture/protocol/index.md:13]: I will have to find more info on this one.

**Pull request merged**